src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Refactoring/RazorSyntaxExtensions.cs

```diff
@@ -0,0 +1,74 @@
+﻿using System;
+using System.Collections.Generic;
+using Microsoft.AspNetCore.Razor.Language;
+using Microsoft.AspNetCore.Razor.Language.Syntax;
+
+namespace Microsoft.AspNetCore.Razor.LanguageServer.Refactoring
+{
+    static class RazorSyntaxExtensions
+    {
+        /// <summary>
+        /// Search for the deepest descendent whose span contains the location and satisfying the filter, if provided.
+        /// Will null if such a node is not found.
+        /// </summary>
+        /// <param name="codeDocument">This code document.</param>
+        /// <param name="location">The location to search for.</param>
+        /// <param name="filter">SyntaxNode constraint.</param>
+        /// <returns>A node at the given location or null.</returns>
+        public static SyntaxNode GetNodeAtLocation(this RazorCodeDocument codeDocument, SourceLocation location, Func<SyntaxNode, bool> filter = null)
```

Comment:
This and `GetChildNodes` aren't used anymore are they? Can we nuke this entire class?